stripe/src/main/java/com/stripe/android/StripeApiHandler.java

```diff
@@ -355,6 +354,84 @@ static String getRetrieveTokenApiUrl(@NonNull String tokenId) {
+     * @return a {@link JSONObject} representing the input map, or {@code null} if the input
+     * object is {@code null}
+     */
+    @Nullable
+    @SuppressWarnings("unchecked")
+    static JSONObject mapToJsonObject(@Nullable Map<String, ? extends Object> mapObject) {
+        if (mapObject == null) {
+            return null;
+        }
+        JSONObject jsonObject = new JSONObject();
+        for (String key : mapObject.keySet()) {
+            Object value = mapObject.get(key);
+            if (value == null) {
+                continue;
+            }
+
+            try {
+                if (value instanceof Map<?, ?>) {
+                    try {
+                        Map<String, Object> mapValue = (Map<String, Object>) value;
+                        jsonObject.put(key, mapToJsonObject(mapValue));
+                    } catch (ClassCastException classCastException) {
+                        // We don't include the item in the JSONObject if the keys are not Strings.
+                    }
+                } else if (value instanceof List<?>) {
+                    jsonObject.put(key, listToJsonArray((List<Object>) value));
+                } else if (value instanceof Number || value instanceof Boolean) {
+                    jsonObject.put(key, value);
+                } else {
+                    jsonObject.put(key, value.toString());
+                }
+            } catch (JSONException jsonException) {
+                // Simply skip this value
+            }
+        }
+        return jsonObject;
+    }
+
+    /**
+     * Converts a {@link List} into a {@link JSONArray}. A {@link ClassCastException} will be
+     * thrown if any object in the list (or any sub-list or sub-map) is a {@link Map} whose keys
+     * are not {@link String Strings}.
+     *
+     * @param values a {@link List} of values to be put in a {@link JSONArray}
+     * @return a {@link JSONArray}, or {@code null} if the input was {@code null}
+     */
+    @Nullable
+    @SuppressWarnings("unchecked")
+    static JSONArray listToJsonArray(@Nullable List values) {
+        if (values == null) {
+            return null;
+        }
+
+        JSONArray jsonArray = new JSONArray();
+        for (Object object : values) {
+            if (object instanceof Map<?, ?>) {
+                try {
+                    Map<String, Object> mapObject = (Map<String, Object>) object;
+                    jsonArray.put(mapToJsonObject(mapObject));
+                } catch (ClassCastException classCastException) {
+                    // We don't include the item in the array if the keys are not Strings.
+                }
+            } else if (object instanceof List<?>) {
+                jsonArray.put(listToJsonArray((List) object));
+            } else if (object instanceof Number || object instanceof Boolean) {
+                jsonArray.put(object);
+            } else {
+                jsonArray.put(object.toString());
+            }
+        }
```

Comment:
This was more necessary before when we didn't have everything package-private -- it might actually be worth getting rid of now. The theory behind is is that we cast an object from `Map` to `Map<String, Object>` in line 419, which breaks [type erasure](https://docs.oracle.com/javase/tutorial/java/generics/erasure.html), one of Java's more infuriating "features." Technically, I can't stop you from feeding a list that contains invalid types here, and when this method was public that was a problem. Making it a private function means it's okay to crash here, since we can control the input.